packages/openapi-fetch/src/index.js

```diff
@@ -84,56 +90,63 @@ export default function createClient(clientOptions) {
       requestInit.headers.delete("Content-Type");
     }
 
-    const id = nanoid();
+    let id;
+    let options;
     let request = new CustomRequest(createFinalURL(schemaPath, { baseUrl, params, querySerializer }), requestInit);
 
-    // middleware (request)
-    const options = Object.freeze({
-      baseUrl,
-      fetch,
-      parseAs,
-      querySerializer,
-      bodySerializer,
-    });
-    for (const m of middlewares) {
-      if (m && typeof m === "object" && typeof m.onRequest === "function") {
-        const result = await m.onRequest({
-          request,
-          schemaPath,
-          params,
-          options,
-          id,
-        });
-        if (result) {
-          if (!(result instanceof Request)) {
-            throw new Error("onRequest: must return new Request() when modifying the request");
+    if (middlewares.length) {
+      id = randomID();
+
+      // middleware (request)
+      options = Object.freeze({
+        baseUrl,
+        fetch,
+        parseAs,
+        querySerializer,
+        bodySerializer,
+      });
+      for (const m of middlewares) {
+        if (m && typeof m === "object" && typeof m.onRequest === "function") {
+          const result = await m.onRequest({
+            request,
+            schemaPath,
+            params,
+            options,
+            id,
+          });
+          if (result) {
+            if (!(result instanceof Request)) {
+              throw new Error("onRequest: must return new Request() when modifying the request");
+            }
+            request = result;
           }
-          request = result;
         }
       }
     }
 
     // fetch!
-    let response = await fetch(request.url, request);
+    let response = await fetch(request);
```

Comment:
This is the central error fix—`Request` can’t be substituted for `RequestInit` in every environment. Trying to hack around this results in even nastier code when you consider different browsers implement different stages of `RequestInit` and experimental APIs under-the-hood. This library shouldn’t take on any of that burden, and instead use a `Request` object which is universal and spec-compliant.